Okta/OktaAuth/OktaAppAuth.swift

```diff
@@ -10,98 +10,105 @@
-}
+    public func signInWithBrowser(from presenter: UIViewController, callback: @escaping ((OktaTokenManager?, OktaError?) -> Void)) {
+        let signInTask = SignInTask(presenter: presenter, config: configuration, oktaAPI: OktaRestApi())
+        currentUserSessionTask = signInTask
 
-public func authenticate(withSessionToken sessionToken: String, callback: @escaping ((OktaTokenManager?, OktaError?) -> Void)) {
-    guard let configuration = configuration else {
-        callback(nil, OktaError.notConfigured)
-        return
+        signInTask.run { [weak self] authState, error in
+            defer { self?.currentUserSessionTask = nil }
+        
+            guard let authState = authState else {
+                callback(nil, error)
+                return
+            }
+            
+            let tokenManager = OktaTokenManager(authState: authState)
+            self?.tokenManager = tokenManager
+            callback(tokenManager, nil)
+        }
     }
 
-    AuthenticateTask(sessionToken: sessionToken, config: configuration, oktaAPI: OktaRestApi())
-    .run { authState, error in
-        guard let authState = authState else {
-            callback(nil, error)
+    public func signOutOfOkta(from presenter: UIViewController, callback: @escaping ((OktaError?) -> Void)) {
+        guard let idToken = tokenManager?.authState.lastTokenResponse?.idToken else {
+            callback(OktaError.missingIdToken)
             return
         }
         
-        let tokenManager = OktaTokenManager(authState: authState)
-        OktaAuth.tokenManager = tokenManager
-        callback(tokenManager, nil)
+        let signOutTask = SignOutTask(idToken: idToken, presenter: presenter, config: configuration, oktaAPI: OktaRestApi())
+        currentUserSessionTask = signOutTask
+        
+        signOutTask.run { [weak self] _, error in
+            self?.currentUserSessionTask = nil
+            callback(error)
+        }
     }
-}
 
-public func clear() {
-    // Clear auth state
-    tokenManager?.clear()
-    tokenManager = nil
-}
+    public func authenticate(withSessionToken sessionToken: String, callback: @escaping ((OktaTokenManager?, OktaError?) -> Void)) {
+        AuthenticateTask(sessionToken: sessionToken, config: configuration, oktaAPI: OktaRestApi())
+        .run { authState, error in
+            guard let authState = authState else {
+                callback(nil, error)
+                return
+            }
+            
+            let tokenManager = OktaTokenManager(authState: authState)
+            self.tokenManager = tokenManager
+            callback(tokenManager, nil)
+        }
+    }
 
-public func getUser(_ callback: @escaping ([String:Any]?, OktaError?) -> Void) {
-    guard let configuration = configuration else {
-        callback(nil, OktaError.notConfigured)
-        return
+    public func clear() {
+        // Clear auth state
+        tokenManager?.clear()
+        tokenManager = nil
```

Comment:
Isn't this a redundant call? `tokenManager.clear` [already sets](https://github.com/okta/okta-sdk-appauth-ios/blob/dev/Okta/OktaAuth/OktaTokenManager.swift#L133) this to `nil`